### PR TITLE
Fix a data race and double-counting of active VUs in arrival-rate executors

### PR DESCRIPTION
**Pull request by na--**
Closes https://github.com/k6io/k6/issues/1954 and fixes https://github.com/k6io/k6/issues/1657

### CODE REVIEW

**Review by MStoykov — APPROVED** [review]
LGTM, good catch :clap:

File: lib/execution.go [hunk 1]
```diff
@@ -558,7 +558,8 @@ func (es *ExecutionState) SetInitVUFunc(initVUFunc InitVUFunc) {
 
 // GetUnplannedVU checks if any unplanned VUs remain to be initialized, and if
 // they do, it initializes one and returns it. If all unplanned VUs have already
-// been initialized, it returns one from the global vus buffer.
+// been initialized, it returns one from the global vus buffer, but doesn't
+// automaticqlly increment the active VUs counter in either case.
```
Comment (imiric) [inline lib/execution.go:562]:
```suggestion
// automatically increment the active VUs counter in either case.
```

**Review by imiric — APPROVED** [review]
Nice one! :clap: Just a typo :smile:

File: lib/execution.go (hunk 1 above)
Comment (na--) [reply lib/execution.go]: Fixed by the latest force push

**Review by MStoykov — APPROVED** [review]

**Review by imiric — APPROVED** [review]

**Pull request merged**